simphony/io/tests/test_serialisation.py

```diff
@@ -0,0 +1,147 @@
+from simphony.cuds.model import CUDS
+from simphony.core.keywords import KEYWORDS
+from simphony.core.cuba import CUBA
+from simphony.io.serialisation import save_CUDS, load_CUDS
+from simphony.cuds.meta.validation import to_camel_case
+
+
+class TestSerialisation(unittest.TestCase):
+    """Tests for CUDS Yaml serialisation functions."""
+    def setUp(self):
+        self.temp_dir = tempfile.mkdtemp()
+
+    def tearDown(self):
+        shutil.rmtree(self.temp_dir)
+
+    def test_save_CUDS_name(self):
+        name = 'somename'
+        filename = os.path.join(self.temp_dir, 'test_named.yml')
+        C = CUDS(name=name)
+        with closing(open(filename, 'w')) as handle:
+            save_CUDS(handle, C)
+
+        with closing(open(filename, 'r')) as handle:
+            CC = load_CUDS(handle)
+        self.assertEqual(CC.name, name)
+
+    def test_save_CUDS_description(self):
+        description = 'some very long description'
+        filename = os.path.join(self.temp_dir, 'test_description.yml')
+        C = CUDS(description=description)
+        with closing(open(filename, 'w')) as handle:
+            save_CUDS(handle, C)
+
+        with closing(open(filename, 'r')) as handle:
+            CC = load_CUDS(handle)
+        self.assertEqual(CC.description, description)
+
+    def test_save_CUDS_empty(self):
+        filename = os.path.join(self.temp_dir, 'test_empty.yml')
+        C = CUDS(name='empty', description='just an empty model')
+        with closing(open(filename, 'w')) as handle:
+            save_CUDS(handle, C)
+
+        with closing(open(filename, 'r')) as handle:
+            CC = load_CUDS(handle)
+
+        self.assertEqual(CC.name, C.name)
+        self.assertEqual(CC.description, C.description)
+
+        for item in CC.iter(CUDSComponent):
+            self.assertEqual(item, None)
+
+    def test_save_CUDS_full(self):
+        filename = os.path.join(self.temp_dir, 'test_full.yml')
+        C = CUDS(name='full', description='fully randomized model')
+
+        for key in KEYWORDS.keys():
+            if key not in ['VERSION']:
+                comp = self._create_random_comp(key)
+                if comp:
+                    C.add(comp)
+
+        with closing(open(filename, 'w')) as handle:
+            save_CUDS(handle, C)
+
+        with closing(open(filename, 'r')) as handle:
+            CC = load_CUDS(handle)
+
+        self.assertEqual(CC.name, C.name)
+        self.assertEqual(CC.description, C.description)
+
+        # Iterate over components in the original model and check
+        # that they are present in the loaded model
+        for Citem in C.iter(CUDSComponent):
+            # Check items that have name parameter defined
+            if Citem.name is not None:
+                CCitem = CC.get(Citem.name)
+                self.assertEqual(CCitem.name, Citem.name)
+                for key in Citem.data:
+                    self.assertEqual(Citem.data[key], CCitem.data[key])
```

Comment:
While trying to fix the failed test I got weired errors this line. Like numpy array errors.